### PR TITLE
[Overflow-30] [Bugfix] Create API for Voting

### PR DESCRIPTION
**Pull request by IanUrriza10**
## Backlog Link
https://framgiaph.backlog.com/view/SUN_OVERFLOW-232
## Commands
```
mutation{
  upsertVote(voteable_id:1, voteable_type:"Answer", value:1)
}
```

## Expected Output
Sending the same value that is voted will delete the vote 1 -> 1
Sending a different value will updated the value 1 -> -1
Sending a new vote will create a new vote 
Added attribute to question and answer to easily access the vote of the user

### CODE REVIEW

**Review by jules-lucero — APPROVED** [review]

File: backend/app/GraphQL/Mutations/UpsertVote.php [hunk 1]
```diff
@@ -27,12 +28,18 @@ public function __invoke($_, array $args)
                 return "Please enter a valid value";
             }
 
-            $voteable->votes()->updateOrCreate(
-                ['user_id'=> Auth::id()],
-                ['value'=>$args['value']]
-            );
+            $current_value = $voteable->votes()->where('user_id',Auth::id())->first()->value ?? 0;
+            if($current_value == $args['value']){
```
Comment (kenorenzsun) [inline backend/app/GraphQL/Mutations/UpsertVote.php]:
```suggestion
            $current_value = $voteable->votes()->where('user_id',Auth::id())->first()->value ?? 0;
            
            if($current_value == $args['value']){
```

File: backend/app/GraphQL/Mutations/UpsertVote.php [hunk 2]
```diff
@@ -27,12 +28,18 @@ public function __invoke($_, array $args)
...
+                $voteable->votes()->where('user_id', Auth::id())->delete();
+                return "Vote Removed";
```
Comment (kenorenzsun) [inline backend/app/GraphQL/Mutations/UpsertVote.php]:
```suggestion
                $voteable->votes()->where('user_id', Auth::id())->delete();
                
                return "Vote Removed";
```

File: backend/app/GraphQL/Mutations/UpsertVote.php [hunk 3]
```diff
@@ -27,12 +28,18 @@ public function __invoke($_, array $args)
...
+            }else{
+                $voteable->votes()->updateOrCreate(
+                    ['user_id'=> Auth::id()],
+                    ['value'=>$args['value']]
+                );
+                return "Voted Successfully";
```
Comment (kenorenzsun) [inline backend/app/GraphQL/Mutations/UpsertVote.php]:
```suggestion
                );
                
                return "Voted Successfully";
```

**Review by kenorenzsun — APPROVED** [review]

File: backend/app/GraphQL/Mutations/UpsertVote.php [hunk 4]
```diff
@@ -29,14 +29,20 @@ public function __invoke($_, array $args)
             }
 
             $current_value = $voteable->votes()->where('user_id',Auth::id())->first()->value ?? 0;
+
             if($current_value == $args['value']){
+
```
Comment (jules-lucero) [inline backend/app/GraphQL/Mutations/UpsertVote.php]: Remove line

File: backend/app/GraphQL/Mutations/UpsertVote.php [hunk 5]
```diff
@@ -29,14 +29,20 @@ public function __invoke($_, array $args)
...
                 $voteable->votes()->where('user_id', Auth::id())->delete();
+
                 return "Vote Removed";
+
```
Comment (jules-lucero) [inline backend/app/GraphQL/Mutations/UpsertVote.php]: Remove line

File: backend/app/GraphQL/Mutations/UpsertVote.php [hunk 6]
```diff
@@ -29,14 +29,20 @@ public function __invoke($_, array $args)
...
             }else{
+
```
Comment (jules-lucero) [inline backend/app/GraphQL/Mutations/UpsertVote.php]: Remove line

**Review by jules-lucero — APPROVED** [review]

**Pull request merged**